src/native/minipal/file.h

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_FILE_H
+#define HAVE_MINIPAL_FILE_H
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <uchar.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif // __cplusplus
+
+typedef struct minipal_file_attr_
+{
+    uint64_t size;
+    uint64_t lastWriteTime; // Windows FILETIME precision
+} minipal_file_attr_t;
+
+bool minipal_file_get_attributes_utf16(const char16_t* path, minipal_file_attr_t* attributes);
```

Comment:
`pal/mstypes.h` defines `WCHAR` as `char16_t`, which is incompatible with `unsigned short` from `minipal/types.h`. Since pal uses `char16_t` unconditionally, I assume it's broad available.

Reply:
> which is incompatible with `unsigned short`

What was the error when you used `CHAR16_T`? You may need to sprinkle a few `(CHAR16_T*)var` at call sites like we did for utf conversion API call sites.

Reply:
Just `char16_t*` and `unsigned short*` is incompatible. Yes there are casts like https://github.com/dotnet/runtime/blob/1aa4568fe91128d2b99b07b978fcf0b11c416e42/src/coreclr/minipal/Unix/dn-u16.cpp#L12

Reply:
On Windows, it is defined as `wchar_t`, while on Unix it is defined as `unsigned short` because `char16_t` is not available everywhere (as seen in current CI errors). I checked various platform headers when defining `CHAR_T` and `unsigned short` was close enough (they use some flexible type but this assumption just works across our supported platforms).